### PR TITLE
Use release.yaml instead of eventing.yaml

### PR DESCRIPTION
**Pull request by grantr**
`eventing.yaml` doesn't install the in-memory provisioner, `release.yaml` does. Since the provisioner is required to run the samples, we should be
telling users to install `release.yaml`.

This (hopefully) fixes the issues people are having running samples with 0.3.0.

### CODE REVIEW

**Comment by samodell** [comment]
Fixes #762 

/lgtm
/approve

**Comment by grantr** [comment]
Just noticed how easy it was to make this mistake: serving uses `serving.yaml` as its release yaml name, so it would make sense if `eventing.yaml` was also the preferred name. However build, eventing, and eventing-sources all use `release.yaml`. We should probably use the same naming scheme for all projects.

**Comment by grantr** [comment]
Also updated `install/Knative-custom-install.md` to use 0.3.0 instead of 0.2.x, and replaced `eventing.yaml` with `release.yaml` in example urls.

**Comment by samodell** [comment]
@grantr  The Knative custom install doc really needs to be rewritten for 0.3.0, which is why I hadn't updated it yet. It has a lot of now obsolete information referring to how the installs used to be bundled together, and makes references to monitoring being included in the Serving install, which is no longer the case. Just updating the links to the 0.3.0 components would cause a lot of confusion. Also, the -lite and -nomon releases for Serving don't exist in 0.3.0.

TL:DR; can you revert your changes to that file? It's on our to-do list to do a thorough review for 0.3.0.

File: install/Knative-custom-install.md [hunk 1]
```diff
@@ -155,23 +155,23 @@ The following Knative installation packages are available:
 
 - **Multiple Component Bundle**:
 
-  - https://github.com/knative/serving/releases/download/v0.2.3/release.yaml
-  - https://github.com/knative/serving/releases/download/v0.2.3/release-no-mon.yaml
-  - https://github.com/knative/serving/releases/download/v0.2.3/release-lite.yaml
+  - https://github.com/knative/serving/releases/download/v0.3.0/release.yaml
```
Comment (samodell) [inline install/Knative-custom-install.md]: Please revert this file. A lot of this info needs to be entirely rewritten for 0.3.0.

Comment (grantr) [reply install/Knative-custom-install.md]: I rebased to remove the commit that changed this file. PTAL!

Comment (samodell) [reply install/Knative-custom-install.md]: thanks! looks good.

**Comment by samodell** [comment]
@evankanderson  The build test failures are a known issue, with a fix in progress. See issue #750 for more background info.
